M_PI is not always available

#### Summary:

The new `log_modified_bessel_first_kind` function utilizes `M_PI` and Windows (at least) manages not to find it.

#### Description:

The new `log_modified_bessel_first_kind` function utilizes `M_PI` and Windows (at least) manages not to find it.


#### Reproducible Steps:

Run unit tests
http://d1m1s1b1.stat.columbia.edu:8080/job/Stan/job/develop/11/execution/node/83/log/

#### Current Output:

```
log_modified_bessel_first_kind.hpp:228:36: error: 'M_PI' was not declared in this scope
           s = z - log(sqrt(2 * z * M_PI)) + log(s);
```

#### Expected Output:

None

#### Additional Information:

#### Current Version:
v2.17.0

--- a/stan/math/prim/scal/fun/log_modified_bessel_first_kind.hpp
+++ b/stan/math/prim/scal/fun/log_modified_bessel_first_kind.hpp
@@ -8,6 +8,7 @@
 #include <stan/math/prim/scal/err/check_not_nan.hpp>
 #include <stan/math/prim/scal/err/check_nonnegative.hpp>
 #include <stan/math/prim/scal/err/check_greater_or_equal.hpp>
+#include <stan/math/prim/scal/fun/constants.hpp>
 #include <stan/math/prim/scal/fun/inv.hpp>
 #include <stan/math/prim/scal/fun/is_inf.hpp>
 #include <stan/math/prim/scal/fun/is_nan.hpp>
@@ -225,7 +226,7 @@ namespace stan {
           num *= mu - 25;
           denom *= ex * 3;
           s -= num / denom;
-          s = z - log(sqrt(2 * z * M_PI)) + log(s);
+          s = z - log(sqrt(2 * z * pi())) + log(s);
           return s;
         }
       }

